subsys/debug/Kconfig

```diff
@@ -181,13 +181,37 @@ config FORCE_NO_ASSERT
 	  CFLAGS and not Kconfig.  Added solely to be able to work
 	  around compiler bugs for specific tests.
 
+config ASSERT_VERBOSE
+	bool "Enable verbose assertions"
+	default y
+	help
+	  This option enables printing an assert message with information about
+	  the assertion that occurred. This includes printing the location,
+	  the conditionial expression and additional message specific to the
+	  assert.
+
 config ASSERT_NO_FILE_INFO
 	bool "Disable file info for asserts"
 	help
 	  This option removes the name and the path of the source file
 	  in which the assertion occurred. Enabling this will save
 	  target code space, and thus may be necessary for tiny targets.
 
+config ASSERT_NO_COND_INFO
+	bool "Disable condition info for asserts"
+	help
+	  This option removes the assert condition from the printed assert
+	  message. Enabling this will save target code space, and thus may be
+	  necessary for tiny targets.
+
+config ASSERT_NO_MSG_INFO
+	bool "Disable message for asserts"
+	help
+	  This option removes the additional message from the printed assert.
+	  Enabling this will save target cod space, and should be disabled
+	  before disabling file and line information as the message can be
```

Comment:
> should be disabled before disabling file and line information

Can we express this dependency in Kconfig itself?

Reply:
Not sure how. But I meant this as a recommendation, and not a rule. I can update the text with "is recommended to" instead of "should"